src/rpc/message_data_structs.h

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2016-2017, The Monero Project
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+// 
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+// 
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#pragma once
+
+#include "crypto/hash.h"
+#include "cryptonote_basic/cryptonote_basic.h"
+#include "ringct/rctSigs.h"
+
+#include <unordered_map>
+#include <vector>
+
+namespace cryptonote
+{
+
+namespace rpc
+{
+
+  struct block_with_transactions
+  {
+    cryptonote::block block;
+    std::unordered_map<crypto::hash, cryptonote::transaction> transactions;
+  };
+
+  typedef std::vector<uint64_t> tx_output_indices;
+
+  typedef std::vector<tx_output_indices> block_output_indices;
+
+  struct transaction_info
+  {
+    cryptonote::transaction transaction;
+    bool in_pool;
+    uint64_t height;
+  };
+
+  struct output_key_and_amount_index
+  {
+    uint64_t amount_index;
+    crypto::public_key key;
+  };
+
+  typedef std::vector<output_key_and_amount_index> outputs_for_amount;
+
+  struct amount_with_random_outputs
+  {
+    uint64_t amount;
+    outputs_for_amount outputs;
+  };
+
+  struct peer
```

Comment:
since the wallet doesn't use that RPC call, I hadn't touched it up yet.  Will deal with that when I go through to check all the other calls I've implemented for changes.